src/sqlitestorage.h

```diff
@@ -672,9 +672,13 @@ class MKCAL_EXPORT SqliteStorage : public ExtendedStorage
 #define SELECT_COMPONENTS_BY_LAST_MODIFIED_AND_NOTEBOOK \
 "select * from Components where DateLastModified>=? and DateCreated<? and Notebook=? and DateDeleted=0"
 #define SELECT_COMPONENTS_BY_DELETED \
-"select * from Components where DateDeleted>=? and DateCreated<?"
+"select * from Components where DateDeleted<>0"
```

Comment:
1) ...will attempt to bind two values to all statements when the operation is DBDelete. This could cause an error: https://www.sqlite.org/rescode.html#range;
As far as I understand, with 'DBDelete' operation, 'SqliteStorage::Private::selectIncidences' currently (with the original code, without my suggestion) binds from 0 (after.isValid() is false, notebookUid.isNull() is true) to 3 parameters (after.isValid() is true, notebookUid.isNull() is false): 2 parameters correspond to 'after' parameter and 1 - to the notebook UID;
Does this make sense, or am I wrong in this respect?